src/west/_bootstrap/main.py

```diff
@@ -109,74 +120,97 @@ def init(argv):
 
     init_parser = argparse.ArgumentParser(
         prog='west init',
-        description='Bootstrap initialize a Zephyr installation')
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=
+'''
+Initializes a Zephyr installation.
+
+In more detail, does the following:
+
+  1. Clones the manifest repository to west/manifest, and the west repository
+     to west/west
+
+  2. Creates a marker file west/{}
+
+  3. Creates an initial configuration file west/config
+
+As an alternative to manually editing west/config, 'west init' can be rerun on
+an already initialized West instance to update configuration settings. Only
+explicitly passed configuration values (e.g. --mr MANIFEST_REVISION) are
+updated.
+
+Updating the manifest URL or revision via 'west init' automatically runs 'west
+update --reset-manifest --reset-projects' afterwards to reset the manifest to
+the new revision, and all projects to their new manifest revisions.
+
+Updating the west URL or revision also runs 'west update --reset-west'.
+
+To suppress the reset of the manifest, west, and projects, pass --no-reset to
+'west init'. With --no-reset, only the configuration file will be updated, and
+you will have to handle any resetting yourself.
+'''.format(WEST_MARKER))
+
     init_parser.add_argument(
         '-b', '--base-url',
-        help='''Base URL for both 'manifest' and 'zephyr' repositories; cannot
-        be given if either -m or -w are''')
+        help='''Base URL for both 'manifest' and 'zephyr' repositories. Cannot
```

Comment:
I'm fine with this change, but FYI, I chose semicolons because none of these help texts end with a period, so I wanted to keep them to a single sentence rather than have them be complete sentences with periods except for the last one.

This is partially to keep consistency with the auto-generated help for the `--help` option.

In general I find it annoying that argparse doesn't capitalize the first words in their help text, but that's a side detail.

Reply:
There was supposed to be a period at the end. Fixed.